app/components/SearchableMetricSelector.js

```diff
@@ -38,17 +75,14 @@ function Modal(props){
               <p>{searchResultsMessage}</p>
             
             </div>
-            <div className="searchResults">
-              {searchResults.map((k, i) => (
-                <p 
-                key={i} 
-                onClick={ () => {
-                  selectMetric(k)
-                  onClose()  
-                } }
-                >{supportedMetrics[k].name}</p>
-              ))}
-            </div>
+            <GroupedMetricSearchResults
+              searchText={searchText} 
+              supportedMetrics={supportedMetrics} 
+              onClose={onClose}
+              selectMetric={selectMetric}
+              tagDictArray={tagDictArray}
+              tagDictionary={tagDictionary}
```

Comment:
Could you move the computation of `tagDictArray` and `tagDictionary` into `GroupedMetricSearchResults`? I don't think it needs to be computed in `Modal`, so it should be an implementation detail of `GroupedMetricSearchResults`.

This will help declutter the `Modal` state variables as well as make `GroupedMetricSearchResults` a true drop-in replacement for `SimpleMetricSearchResults`.